examples/Planning/ForecastReach.cs

```diff
@@ -0,0 +1,313 @@
+        /// Create a base request to generate a reach forecast.
+        /// </summary>
+        /// <param name="customerId">The customer ID for the reach forecast.</param>
+        /// <param name="productMix">The product mix for the reach forecast.</param>
+        /// <param name="locationId">Location ID from somewhere returned by ListPlannableLocations.</param>
+        /// <param name="currencyCode">Three-character ISO 4217 currency code.</param>
+        public GenerateReachForecastRequest BuildReachRequest(
+                   string customerId,
+                   List<PlannedProduct> productMix,
+                   string locationId,
+                   string currencyCode)
+        {
+            CampaignDuration duration = new CampaignDuration()
+            {
+                DurationInDays = 30
+            };
+
+            GenderInfo[] genders = new GenderInfo[]
+            {
+                new GenderInfo() { Type = GenderType.Female },
+                new GenderInfo() { Type = GenderType.Male }
+            };
+
+            DeviceInfo[] devices = new DeviceInfo[]
+            {
+                new DeviceInfo() { Type = Device.Desktop },
+                new DeviceInfo() { Type = Device.Mobile },
+                new DeviceInfo() { Type = Device.Tablet }
+            };
+
+            Targeting targeting = new Targeting()
+            {
+                PlannableLocationId = locationId,
+                AgeRange = ReachPlanAgeRange.AgeRange1865Up,
+
+            };
+            targeting.Genders.AddRange(genders);
+            targeting.Devices.AddRange(devices);
+
+            GenerateReachForecastRequest request = new GenerateReachForecastRequest()
+            {
+                CustomerId = customerId,
+                CurrencyCode = currencyCode,
+                CampaignDuration = duration,
+                Targeting = targeting,
+                MinEffectiveFrequency = 3
+            };
+
+            request.PlannedProducts.AddRange(productMix);
+
+            return request;
+        }
+
+        /// <summary>
+        /// Pulls and prints the reach curve for the given request.
+        /// </summary>
+        /// <param name="reachPlanService">Instance of Reach Plan Service client.</param>
+        /// <param name="request">An already-populated reach curve request.</param>
+        public void PullReachCurve(ReachPlanServiceClient reachPlanService, GenerateReachForecastRequest request)
+        {
+            GenerateReachForecastResponse response = reachPlanService.GenerateReachForecast(request);
+            Console.WriteLine("Reach curve output:");
+            Console.WriteLine(
+                "Currency,\tCost,\tOn-Target Reach,\tOn-Target Imprs,\tTotal Reach,\tTotal Imprs,\tProducts");
+            foreach (ReachForecast point in response.ReachCurve.ReachForecasts)
+            {
+                Console.Write($"{request.CurrencyCode},\t");
+                Console.Write($"{point.CostMicros / 1.0e6},\t");
```

Comment:
API uses cost in micros, so please print it as such without converting to currency values. You can make the title Cost (Micros) to keep things clearer/

Reply:
Done